### ISSUE TITLE
Request to add custom providers like Azure, GCP, etc

### ISSUE THREAD

**Issue by Eskibear** [awslabs/git-secrets#194]
Also mentioned in #150 #171 

I see forks (e.g. https://github.com/deshpandetanmay/git-secrets and https://github.com/msalemcode/git-secrets) adding GCP and Azure patterns, here I just want to confirm whether you are open to custom providers, and we can definitely provide patterns and contribute.

**Comment by chathudan** [comment]
Can this merge?

**Comment by sparr** [comment]
I agree that including sane defaults for these other platforms would be valuable, especially when they are already implemented in active forks.
